app/controllers/items_controller.rb

```diff
@@ -45,6 +47,21 @@ def item_params
 
   private
 
+  def set_show_headline(item, review_count)
```

Comment:
Nice method, but really busy. Move some of the logic into that empty `calculate_average_review` method you have below to abstract the logic. OK for now.